yellowbrick/features/scatter.py

```diff
@@ -0,0 +1,286 @@
+            Matplotlib style markers for points on the scatter plot points
+
+        kwargs : keyword arguments passed to the super class.
+
+        These parameters can be influenced later on in the visualization
+        process, but can and should be set as early as possible.
+    """
+
+    def __init__(self,
+                 ax=None,
+                 features=None,
+                 classes=None,
+                 color=None,
+                 colormap=None,
+                 markers=None,
+                 **kwargs):
+        """
+        Initialize the base scatter with many of the options required in order
+        to make the visualization work.
+        """
+        super(ScatterVisualizer, self).__init__(ax, features, classes, color,
+                                                colormap, **kwargs)
+        self.markers = itertools.cycle(
+            kwargs.pop('markers', (',', '+', 'o', '*', 'v', 'h', 'd')))
+
+        # Ensure with init that features doesn't have more than two features
+        if features is not None:
+            if len(features) != 2:
+                raise YellowbrickValueError(
+                    'ScatterVisualizer only accepts two features')
+
+    def fit(self, X, y=None, **kwargs):
+        """
+        The fit method is the primary drawing input for the parallel coords
+        visualization since it has both the X and y data required for the
+        viz and the transform method does not.
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with 2 features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer
+        """
+        _, ncols = X.shape
+
+        if ncols == 2:
+            X_two_cols = X
+            if self.features_ is None:
+                self.features_ = ["Feature One", "Feature Two"]
+
+        # Handle the feature names if they're None.
+        elif self.features_ is not None and is_dataframe(X):
+            X_two_cols = X[self.features_].as_matrix()
+
+        # TODO handle numpy named arrays
+        # handle numpy named/ structured array
+        # elif self.features_ is not None and is_structured_array(X):
+        #     X_two_cols = X[self.features_]
+
+        else:
+            raise YellowbrickValueError("""
+                ScatterVisualizer only accepts two features, please
+                explicitly set these two features in the init kwargs or
+                pass a matrix/ dataframe in with only two columns.""")
+
+        # Store the classes for the legend if they're None.
+        if self.classes_ is None:
+            # TODO: Is this the most efficient method?
+            self.classes_ = [str(label) for label in np.unique(y)]
```

Comment:
Ok, cool - glad to know that's faster; I might run some benchmarks to see this in action; I'll let you know when I do them.